src/HIE/Bios/Cradle.hs

```diff
@@ -493,37 +519,57 @@ withCabalWrapperTool (mbGhc, ghcArgs) wdir = do
 -- | Given the root directory, get the build dir we are using for cabal
 -- In the `hie-bios` cache directory
 cabalBuildDir :: FilePath -> IO FilePath
-cabalBuildDir work_dir = do
-  abs_work_dir <- makeAbsolute work_dir
+cabalBuildDir workDir = do
+  abs_work_dir <- makeAbsolute workDir
   let dirHash = show (fingerprintString abs_work_dir)
-  getCacheDir ("dist-"<>filter (not . isSpace) (takeBaseName abs_work_dir)<>"-"<>dirHash)
+  getCacheDir ("dist-" <> filter (not . isSpace) (takeBaseName abs_work_dir)<>"-"<>dirHash)
+
+cabalGHCDir :: FilePath -> IO (CradleLoadResult FilePath)
+cabalGHCDir workDir = do
+  readProcessWithCwd workDir  "cabal" ["exec", "-v0", "--", "ghc", "--print-libdir"] ""
+    >>= \case
+      CradleSuccess r -> do
+        let strippedPath = T.unpack $ T.stripEnd $ T.pack r
+        pure $ CradleSuccess $
+          if isWindows
+            then strippedPath </>"../bin"
+            else strippedPath </> "../../bin"
```

Comment:
I suggested that this should be a recursive search function, e.g. look in every parent directory whether there is a `bin` and see if it contains what we need. 
Your opinion @jneira, or whoever read this?

Reply:
ghc itself [does a recursive function with a max depth of three](https://gitlab.haskell.org/ghc/ghc/-/blob/master/compiler/GHC/SysTools/BaseDir.hs#L184-203) to discover its mingw internal 😄 

```haskell
findToolDir
  :: FilePath -- ^ topdir
  -> IO (Maybe FilePath)
#if defined(mingw32_HOST_OS) && !defined(USE_INPLACE_MINGW_TOOLCHAIN)
findToolDir top_dir = go 0 (top_dir </> "..") []
  where maxDepth = 3
        go :: Int -> FilePath -> [FilePath] -> IO (Maybe FilePath)
        go k path tried
          | k == maxDepth = throwGhcExceptionIO $
              InstallationError $ "could not detect mingw toolchain in the following paths: " ++ show tried
          | otherwise = do
              let try = path </> "mingw"
              let tried = tried ++ [try]
              oneLevel <- doesDirectoryExist try
              if oneLevel
                then return (Just path)
                else go (k+1) (path </> "..") tried
#else
findToolDir _ = return Nothing
#endif
```
so this version is fine as is, i guess